lychee-lib/src/filter/mod.rs

```diff
@@ -1,72 +1,140 @@
+    ///    - When both excludes and includes rules are empty, it's *presumably included* unless
+    ///      it's a known false positive.
+    ///    - When the includes rules matches the URI, it's *explicitly included*.
+    /// 3. When it's a known *false positive* pattern, it's *explicitly excluded*.
+    /// 4. Decide whether the URI is *presumably excluded* or *explicitly excluded*:
+    ///    - When excludes rules is empty, but includes rules doesn't match the URI, it's
+    ///      *presumably excluded*.
+    ///    - When the excludes rules matches the URI, it's *explicitly excluded*.
     pub fn is_excluded(&self, uri: &Uri) -> bool {
-        // Skip mail?
-        if self.excludes.is_mail_excluded() && uri.scheme() == "mailto" {
-            return true;
-        }
-        // Skip specific IP address?
-        if self.excludes.ip(&uri) {
+        // Skip mail address, specific IP, and scheme
+        if self.is_mail_excluded(uri) || self.is_ip_excluded(uri) || self.is_scheme_excluded(uri) {
             return true;
         }
 
         let input = uri.as_str();
-        if self.includes.is_empty() {
-            if self.excludes.is_empty() {
-                // No regex includes/excludes at all?
-                // Not excluded unless it's a known false positive
-                return Excludes::is_false_positive(input);
+
+        if self.is_includes_empty() {
+            if self.is_excludes_empty() {
+                // Both excludes and includes rules are empty:
+                // *Presumably included* unless it's false positive
+                return is_false_positive(input);
             }
-        } else if self.includes.regex(input) {
-            // Included explicitly (Includes take precedence over excludes)
+        } else if self.is_includes_match(input) {
+            // *Explicitly included* (Includes take precedence over excludes)
             return false;
         }
-        // Exclude well-known false-positives.
-        // This is done after checking includes to allow for user-overwrites.
-        if Excludes::is_false_positive(uri.as_str()) {
-            return true;
-        }
-        if self.excludes.is_empty() {
-            if !self.includes.is_empty() {
-                // In case we have includes and no excludes,
-                // skip everything that was not included
-                return true;
-            }
-        } else if self.excludes.regex(input) {
-            // Excluded explicitly
+
+        if is_false_positive(input)
+        // Exclude well-known false-positives
+        // Performed after checking includes to allow user-overwriddes
+            || self.is_excludes_empty()
+        // Previous checks implies input is not explicitly included,
+        // if excludes rules is empty, then *presumably excluded*
+            || self.is_excludes_match(input)
+        // If excludes rules matches input, then
+        // *explicitly excluded*
+        {
```

Comment:
In the above if conditions, I'd personally also move the comment before the condition. At least I find it easier to read this way, but it's subjective of course. Just wonder if `rustfmt` would also change the indentation in that case.

Reply:
```rust
        if
        // Exclude well-known false-positives
        // Performed after checking includes to allow user-overwriddes
        is_false_positive(input)
                // Previous checks imply input is not explicitly included,
                // if excludes rules is empty, then *presumably excluded*
                || self.is_excludes_empty()
                // If excludes rules matches input, then
                // *explicitly excluded*
            || self.is_excludes_match(input)
        {
            return true;
        }
```

This is also hard to read